engine/src/mblstore.cpp

```diff
@@ -141,14 +154,8 @@ bool MCPurchaseStateToString(MCPurchaseState p_state, const char *&r_string)
 }
 
 bool MCPurchaseList(MCStringRef& r_string)
-{   
-	MCAutoListRef t_list;
-	if (!MCListCreateMutable('\n', &t_list))
-		return false;
-	for (MCPurchase *t_purchase = MCStoreGetPurchases(); t_purchase != NULL; t_purchase = t_purchase->next)
-        MCListAppendFormat(*t_list,"%@", t_purchase -> prod_id); // we want a list of product IDs
-	
-	return MCListCopyAsString(*t_list, r_string);
+{
+    return MCListCopyAsString(s_completed_purchases, r_string);
```

Comment:
This will crash if s_completed_purchases == nil (i.e. if the user has not made/restored any purchases). In that case, we want to return true and assign empty to r_string